### DISCUSSION TITLE
Does Zephyr project provide sample code for sending battery voltage through Bluetooth battery service?

### DISCUSSION THREAD

**Discussion by snowuyl** [zephyrproject-rtos/zephyr#61883] [General] [answered]
I would like to support sending battery voltage through Bluetooth battery service. Does Zephyr project provide sample code for sending battery voltage through Bluetooth battery service?

**Comment by snowuyl** [comment]
[BAS_SPEC_V10.pdf](https://github.com/zephyrproject-rtos/zephyr/files/12437013/BAS_SPEC_V10.pdf)

**Comment by nordicjm** [comment, marked as answer]
BAS is supported out of the box in zephyr yes, you can see how to use it in some of the sample applications: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/peripheral_csc/src/main.c#L386 though you would need to calculate the level that it shows to a client with your own code as that would be device specific (the sample above sends a dummy changing value)
